flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/KnnModelData.java

```diff
@@ -0,0 +1,134 @@
+package org.apache.flink.ml.classification.knn;
+
+import org.apache.flink.api.common.serialization.Encoder;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.connector.file.src.reader.SimpleStreamFormat;
+import org.apache.flink.core.fs.FSDataInputStream;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.types.Row;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/** knn model data, which will be used to calculate the distances between nodes. */
+public class KnnModelData implements Serializable, Cloneable {
+    private static final long serialVersionUID = -2940551481683238630L;
```

Comment:
Like the comments said in NaiveBayes PR, maybe we don't need to set this `serialVersionUID`. And that it might be better to serialize model data into primitives, instead of java-specific objects.

Reply:
OK, I will refine it later.

Reply:
done